File: lkc_frontend/lkc/lib/models/extension_board.dart
import 'package:lkc/models/product.dart';

class ExtensionBoard extends Product {
  ExtensionBoard(
      {String id,
      String brand,
      String color,
      String image,
      String model,
      double price,
      bool top})
      : super(
            brand: brand,
            id: id,
            color: color,
            image: image,
            model: model,
            price: price,
            top: top);

  factory ExtensionBoard.fromJson(Map<String, dynamic> map) {
    return ExtensionBoard(
      id: map['id'],
      brand: map['brand'],
      color: map['color'],
      image: map['image'],
      model: map['model'],
      price: map['price'],
      top: map['top']
    );
  }
}

File: lkc_backend/lkc/src/main/java/com/lkc/lkc/models/ExtensionBoard.java
package com.lkc.lkc.models;

import org.springframework.data.annotation.TypeAlias;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "product")
@TypeAlias("extensionBoard")
public class ExtensionBoard extends Product {

    public ExtensionBoard(String brand, String color, String id, String image, String model, double price, boolean top,
            String type, double quantity, int numberOfRatings, int rating) {
        super(brand, color, id, image, model, price, top, type, quantity, numberOfRatings, rating);
    }

}
